src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LoggerAdapter.cs

```diff
@@ -150,10 +149,10 @@ public void LogWarning(string message, params object[] @params)
 #pragma warning restore CA2254 // Template should be a static expression
     }
 
-    private class CompositeDisposable : IDisposable
+    private sealed class CompositeDisposable : IDisposable
     {
         private bool _disposed = false;
-        private readonly IList<IDisposable> _disposables = new List<IDisposable>();
+        private readonly List<IDisposable> _disposables = [];
```

Comment:
📝 By changing this from an `IList` to a `List`, we avoid boxing a struct-based enumerator in the foreach below.